tests/test_generic.py

```diff
@@ -768,3 +769,306 @@ def test_spell_length_statistics_multi(tasmin_series, tasmax_series):
+
+        np.testing.assert_array_equal(out.event_length, [3, 3, 7])
+        np.testing.assert_array_equal(out.event_effective_length, [3, 3, 4])
+        np.testing.assert_array_equal(out.event_sum, [16, 6, 10])
+        np.testing.assert_array_equal(
+            out.event_start,
+            np.array(
+                ["2000-01-08", "2000-01-17", "2000-01-27"], dtype="datetime64[ns]"
+            ),
+        )
+
+    @pytest.mark.parametrize("use_dask", [True, False])
+    def test_cftime(self, pr_series, use_dask):
+        arr = np.array(
+            [
+                0,
+                0,
+                0,
+                1,
+                2,
+                3,
+                0,
+                3,
+                3,
+                10,
+                0,
+                0,
+                0,
+                0,
+                0,
+                1,
+                2,
+                2,
+                2,
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+                1,
+                3,
+                3,
+                2,
+                0,
+                0,
+                0,
+                2,
+                0,
+                0,
+                0,
+                0,
+            ]
+        )  # noqa
+        pr = pr_series(arr, start="2000-01-01", units="mm").convert_calendar("noleap")
+        if use_dask:
+            pr = pr.chunk(-1)
+
+        # cftime
+        with assert_lazy:
+            out = generic.thresholded_events(
+                pr,
+                thresh="1 mm",
+                op=">=",
+                window=3,
+            )
+        out = out.load().dropna("event", how="all")
+
+        np.testing.assert_array_equal(out.event_length, [7, 4, 4])
+        np.testing.assert_array_equal(out.event_effective_length, [6, 4, 4])
+        np.testing.assert_array_equal(out.event_sum, [22, 7, 9])
+        exp = xr.DataArray(
+            [1, 2, 3],
+            dims=("time",),
+            coords={
+                "time": np.array(
+                    ["2000-01-04", "2000-01-16", "2000-01-26"], dtype="datetime64[ns]"
+                )
+            },
+        )
+        np.testing.assert_array_equal(
+            out.event_start, exp.convert_calendar("noleap").time
+        )
+
+    @pytest.mark.parametrize("use_dask", [True, False])
+    def test_freq(self, pr_series, use_dask):
+        jan = [
+            0,
+            0,
+            0,
+            1,
+            2,
+            3,
+            0,
+            3,
+            3,
+            10,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            2,
+            2,
+            2,
+            2,
+            2,
+            2,
+            0,
+            0,
+            0,
+            0,
+            0,
+            3,
+            2,
+            3,
+            2,
+        ]  # noqa
```

Comment:
is it worth to make the notation less explicit but more compact in this case?
e.g.
```suggestion
        jan = [0]*3+[1,2, 3, 0, 3, 3,10] + [0]*6 + [2]*6 + [0]*5 + [3, 2, 3, 2]]  # noqa
```

Reply:
dang the `#noqa` didn't work as expected... You idea works better!

Reply:
hmmm if you can make the #noqa work don't bother rewritting everything

Reply:
```
some_really_long_line = [..., ..., ...]  # fmt: skip
```